fastly/block_fastly_service_v1_logging_heroku_test.go

```diff
@@ -0,0 +1,235 @@
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+		Token:             "secret",
+		FormatVersion:     2,
+		Format:            "%h %l %u %t \"%r\" %>s %b %T",
+	}
+
+	log2 := gofastly.Heroku{
+		Version:       1,
+		Name:          "another-heroku-endpoint",
+		URL:           "https://new.example.com",
+		Token:         "another-token",
+		FormatVersion: 2,
+		Format:        "%h %l %u %t \"%r\" %>s %b",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1HerokuConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1HerokuAttributes(&service, []*gofastly.Heroku{&log1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_heroku.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1HerokuConfig_update(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1HerokuAttributes(&service, []*gofastly.Heroku{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_heroku.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1HerokuAttributes(service *gofastly.ServiceDetail, heroku []*gofastly.Heroku) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		herokuList, err := conn.ListHerokus(&gofastly.ListHerokusInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up Heroku Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(herokuList) != len(heroku) {
+			return fmt.Errorf("Heroku List count mismatch, expected (%d), got (%d)", len(heroku), len(herokuList))
+		}
+
+		log.Printf("[DEBUG] herokuList = %#v\n", herokuList)
+
+		var found int
+		for _, e := range heroku {
+			for _, el := range herokuList {
+				if e.Name == el.Name {
+					// we don't know these things ahead of time, so populate them now
+					e.ServiceID = service.ID
+					e.Version = service.ActiveVersion.Number
+					// We don't track these, so clear them out because we also wont know
+					// these ahead of time
+					el.CreatedAt = nil
+					el.UpdatedAt = nil
+					if diff := cmp.Diff(e, el); diff != "" {
+						return fmt.Errorf("Bad match Heroku logging match: %s", diff)
+					}
+					found++
+				}
+			}
+		}
+
+		if found != len(heroku) {
```

Comment:
This seems like a logical duplicate of https://github.com/terraform-providers/terraform-provider-fastly/pull/271/files#diff-5fc436950cb3d96e1cf332ee53bd23a5R134.